spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -76,28 +74,12 @@ public CompletableFuture<MessageId> sendAsync(String topic, T message, MessageRo
 						this.logger.error(ex, () -> String.format("Failed to send msg to '%s' topic", topicName));
 						// TODO fail metrics
 					}
-					closeProducerAsync(producer);
+					ProducerUtils.closeProducerAsync(producer, this.logger);
```

Comment:
I am going to follow on w/ another update to PulsarTemplateTests if need be to ensure that close is being called now that this has been added